lib/pages/locations.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../data/bloc/rick_and_morty_bloc.dart';
+import '../data/models/locations.dart';
+import '../widgets/location_item.dart';
+import '../widgets/preloading_indicator.dart';
+
+class LocationsPage extends StatefulWidget {
+  @override
+  _LocationsPageState createState() => _LocationsPageState();
+}
+
+class _LocationsPageState extends State<LocationsPage> {
+  final List<Locations> _list = [];
+
+  bool newPageError = false;
+
+  @override
+  void initState() {
+    super.initState();
+    newPageError = false;
```

Comment:
now because your state doesn't have "historical" info, you have to do the hack with extra variables in multiple places